clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerPartitionAssignor.java

```diff
@@ -44,7 +44,9 @@
      * Return serialized data that will be included in the {@link Subscription} sent to the leader
      * and can be leveraged in {@link #assign(Cluster, GroupSubscription)} ((e.g. local host/rack information)
      *
-     * @return optional join subscription user data
+     * @param topics Topics subscribed to through {@link org.apache.kafka.clients.consumer.KafkaConsumer#subscribe(java.util.Collection)}
+     *               and variants
+     * @return optional subscription user data
      */
     default ByteBuffer subscriptionUserData(Set<String> topics) {
```

Comment:
By the way, what do you think about making the return type `Optional`?

Reply:
I'm not really a fan of it here, seems like it just adds noise when it's obviously optional given you have to explicitly override this to return anything. But I don't feel too strongly so it can be made Optional if that's preferred

Reply:
Yeah, I don't feel strongly about it either. We can leave it as is. Maybe the javadoc could describe it as "nullable" instead of "optional" at least?

Reply:
Fair enough